stdlib/http/src/main/ballerina/http/service_endpoint.bal

```diff
@@ -164,13 +230,183 @@ public type KeepAlive "AUTO"|"ALWAYS"|"NEVER";
 @final public KeepAlive KEEPALIVE_NEVER = "NEVER";
 
 function Listener::init (ServiceEndpointConfiguration c) {
+    match c.authProviders {
+        AuthProvider[] providers => addAuthFiltersForSecureListener(c, self.instanceId);
+        () => {}
+    }
     self.config = c;
```

Comment:
Shall we move this statement as the first statement of the init method and use self.config.authProviders in the match?